src/lib/s3Uploader.js

```diff
@@ -0,0 +1,45 @@
+import { S3 } from 'aws-sdk';
+
+export const s3 = new S3({
+  accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+  endpoint: process.env.S3_ENDPOINT,
+  secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+  signatureVersion: 'v4',
+  s3ForcePathStyle: true,
+});
+
+export const verifyVersioning = async (bucket = process.env.bucket, s3Client = s3) => {
+  const versioningConfiguration = {
+    MFADelete: 'Disabled',
+    Status: 'Enabled',
+  };
+  let params = {
+    Bucket: bucket,
+  };
+  const data = await s3Client.getBucketVersioning(params);
+  if (!(data) || data.Status !== 'Enabled') {
+    params = {
+      Bucket: bucket,
+      VersioningConfiguration: versioningConfiguration,
+    };
+    return s3Client.putBucketVersioning(params);
+  }
+  return new Promise((resolve) => resolve(data));
+};
+
+const s3Uploader = async (buffer, name, type, s3Client = s3) => {
+  const params = {
+    Body: buffer,
+    Bucket: process.env.S3_BUCKET,
+    ContentType: type.mime,
+    Key: name,
+  };
+  // Only check for versioning if not using Minio
+  if (!process.env.S3_ENDPOINT.includes('minio')) {
```

Comment:
Should we add `S3_ENDPOINT` to the `.env`? If so, with what value?

Reply:
If you are running docker, then that works. If not, http://localhost:9000 should work. However, it will probably through an error since minio doesn't support versioning. Maybe this line should be updated to 
```
if (!process.env.S3_ENDPOINT.includes('minio') ||  !process.env.S#_ENDPOINT.includes('localhost')) {
//do something
}
```
Or we could add a .env variable like LOCAL_DEV and have that be true whenever using minio, then we wouldn't need to worry about what the endpoint was set to.

Reply:
I just pushed an update.
Pull the latest and set
`S3_ENDPOINT="http://localhost:9000"`
and
`LOCAL_DEV=true` in your .env file. That should work for non-docker development.